test/libyul/DominatorTest.cpp

```diff
@@ -90,7 +90,7 @@ class DominatorFixture
 		{
 			return {_pair.first.name, _pair.second};
 		};
-		return _vertexIndices | ranges::views::transform(convertIndex) | ranges::to<std::map>;
```

Comment:
For some reason xcode does not compile when using `ranges::to<std::map>`, see: https://app.circleci.com/pipelines/github/ethereum/solidity/31124/workflows/a8973a83-9fa8-411b-aa4b-c9f2adddb418/jobs/1385058/parallel-runs/0/steps/0-103

It seems that it cannot infer the correct type of the container to perform the conversion. It works on Linux and Windows, but not on MacOS builds.

Reply:
I have one more weird macOS issue.

I thought I'd simplify this:
```c++
ranges::for_each(
	bucket[wIdx],
	[&](size_t vIdx)
	{
		size_t uIdx = eval(vIdx);
		solAssert(uIdx <= vIdx);
		idom[vIdx] = (semi[uIdx] < semi[vIdx]) ? uIdx : wIdx;
	}
);
```

into this:
```c++
for (size_t vIdx: bucket[wIdx])
{
	size_t uIdx = eval(vIdx);
	solAssert(uIdx <= vIdx);
	idom[vIdx] = (semi[uIdx] < semi[vIdx]) ? uIdx : wIdx;
}
```

But apparently [it causes an access violation on macOS](https://app.circleci.com/pipelines/github/ethereum/solidity/33780/workflows/ae2584ae-1651-4552-a61c-afa0c60e6bb1/jobs/1524778):

```
errors disabling the alternate stack:
	#error:22
	Invalid argument
unknown location:0: fatal error: in "Dominators/idfsquad": memory access violation at address: 0x169ae3f20: invalid permissions
/solidity/test/libyul/DominatorTest.cpp:454: last checkpoint: "idfsquad" test entry
```

Can't really see why. Is there actually a good reason to use `for_each()` instead of plain `for` here that I don't see?

Reply:
> Can't really see why. Was there actually a good reason to use `for_each()` instead of plain `for` here that I don't see?

Yeah, I had the same problem hahaha. This is why I used `for_each()`. I also don't know why it doesn't work on MacOS.

Reply:
ok, fair enough. Though I'm a little worried that the workaround might just be hiding the issue rather than safely working around it. I'd feel better about it if I understood why the plain version segfaults. It could be just a random effect of a bug that's actually elsewhere in the algorithm.

Reply:
True. I'm honestly afraid of the same. This is why I commented here about the strange behavior on the MacOS builds, because it it is very possible that there is some problem in the algorithm that we are not seeing. I will run some tests with the AddressSanitizer to see if I can catch some error.

Reply:
Indeed there is a memory leak, apparently in the `TestVertex` when allocating the `successors` due to cyclic dependencies between the shared pointers used by `DominatorFinderTest.vertices` and `TestVertex.successors`. Running the tests with the address sanitizer reported the problem.

Using weak pointers in the `TestVertex` fixed the issue by allowing the shared pointers to be deallocated when there are not more references to them.